### PR TITLE
Hivemind new ability: Nightfall

### PR DESCRIPTION
**Pull request by BraveMole**
## About The Pull Request

Hivemind get a new ability, nightfall. Every two minutes, will shut down all lights around the turf targeted for 5 seconds. You won't be able able to turn them on again during that time, but they will turn back on automaticly after these 5 seconds.

Will shut down the following lights :

- flashlight
- flares (permanently)
- hard hats
- suit lights
- light fixtures
- floodlights
- flashlights on gun

So the only way for marines to see anything, will be using a flamer, or the muzzle flash

## Why It's Good For The Game

Makes hivemind more usefull in combat, by adding a feature that i think can be fun for marines. Will tweak if busted

## Changelog
:cl:
add: New ability for hivemind, nightfall. Will shutdown all lights around the targeted zone for 5 seconds
/:cl:

### CODE REVIEW

**Comment by tgstation-server** [comment]
You currently have a negative Fix/Feature pull request delta of -5. Maintainers may close this PR at will. Fixing issues or improving the codebase will improve this score.

**Comment by Chaznoodles** [comment]
>Make hivemind even more annoying than it currently is

I don't know how you managed it, but you managed it. Not a good or interesting ability.

**Comment by BraveMole** [comment]
> > Make hivemind even more annoying than it currently is
> 
> I don't know how you managed it, but you managed it. Not a good or interesting ability, equivalent to having a free 'blind these people' button on a caste that you can't fight back against.

What a surprise, you critisizing all my PRs. Eh

**Comment by Smudged** [comment]
I'm not too crazy on an ability that well, is oppressive and that marines can't interact with.

Also, a 10 second light shut off on a 2 minute cool down is waaaaay too long, I'd recommend tuning it down to 5 or even 3 seconds.  

Until hivemind is a little more different or more easily  attacked by marines by virtue of needing to be "there" during a fight, I don't think this ability, right now, would be healthy for hivemind.

**Comment by BraveMole** [comment]
> I'm not too crazy on an ability that well, is oppressive and that marines can't interact with.
> 
> Also, a 10 second light shut off on a 2 minute cool down is waaaaay too long, I'd recommend tuning it down to 5 or even 3 seconds.
> 
> Until hivemind is a little more different or more easily attacked by marines by virtue of needing to be "there" during a fight, I don't think this ability, right now, would be healthy for hivemind.

Yeah, tuned down to 5 sec.  I agree, would probably be much better with the hivemind rework, but since it's a soonTM PR, i want to test this without it before

**Comment by Losenis** [comment]
I'd have to agree on hivemind being an annoyance, it can creep towards sentries with walls and sacs without any danger to itself other than wasted time, but I'll wait to see on how this plays out. It does not remove any agency from players unlike stuns and it won't be as effective in long narrow hallways where there's one way to shoot at, but it might help xenos deal with larger groups in open areas. It's not like hivemind can kill marines, so it might be worthwhile in exchange for a player that could've brought a warrior/rav/shrike/any other caste to the front being effectively removed from direct combat.

**Comment by Sulaboy** [comment]
I hope the projectile light still goes, and I honestly can’t remember if muzzle flash makes light here or if it’s just bullets, and if muzzle flash does make light I’m fairly certain it’s a very small area around your character.

I like having electronic lights turn off (APCs, armor lights, spotlights), but flares, and muzzle flashes should still make light. If muzzle flash light range was increased (Still below suit light range) it would help nightfall in what it feels like it’s going for. Setting up an ambush rather than using a blind during a fight.  It could be good for the feel of the game if you can get it done right, and I like the idea of fighting through the light of your muzzle flash.

Also it should have a cool sound effect
https://youtu.be/4qDeioLBiao

**Comment by BraveMole** [comment]
> I like having electronic lights turn off (APCs, armor lights, spotlights), but flares, and muzzle flashes should still make light. If muzzle flash light range was increased (Still below suit light range) it would help nightfall in what it feels like it’s going for. Setting up an ambush rather than using a blind during a fight. It could be good for the feel of the game if you can get it done right, and I like the idea of fighting through the light of your muzzle flash.

I thought of letting flares lit up, and i'm still not sure if they should be affected or not. This is one of the variable I can change later

Muzzle flash is not affected

**Comment by Losenis** [comment]
Where do flashlights fall into?

**Comment by Sulaboy** [comment]
> I thought of letting flares lit up, and i'm still not sure if they should be affected or not. This is one of the variable I can change later
> 
> Muzzle flash is not affected

If I’m recalling correctly muzzle flash was put in on cm for HvH to help see who’s shooting you in the dark. If you buff the light range to something workable like a unmodded jaeger sit lamp marines are still put at a large disadvantage, but have some counter play to it. The counter being that as long as marines are shooting they would have vision albeit limited. If the hivemind uses nightfall during an engagement it would be less effective than in an ambush.

I can see this ability being very good thematically for the game. Walking with your squad when the lights go off, you hear slashing and a scream, the marine starts shooting and you now see your attackers.

Either way cool idea man.

File: code/modules/mob/living/carbon/xenomorph/castes/hivemind/abilities_hivemind.dm [hunk 1]
```diff
@@ -5,3 +5,26 @@
 
 /datum/action/xeno_action/return_to_core/action_activate()
 	SEND_SIGNAL(owner, COMSIG_XENOMORPH_CORE_RETURN)
+
+/datum/action/xeno_action/activable/nightfall
+	name = "Nightfall"
+	action_icon_state = "nightfall"
+	mechanics_text = "Shut down all light for 10 seconds"
```
Comment (Moggram) [inline code/modules/mob/living/carbon/xenomorph/castes/hivemind/abilities_hivemind.dm:12]: mechanics_text = "Shut down all light for 5 seconds


I think its a cool ability and can not wait for that oh shit moment when my lights die.

**Comment by SplinterGP** [comment]
not plausible until hivemind gets a rework of some sorts, hivemind is omnipresent invencible thing so adding anything to it as is is bad

**Comment by BraveMole** [comment]
> not plausible until hivemind gets a rework of some sorts, hivemind is omnipresent invencible thing so adding anything to it as is is bad

Of some sort :)

**Comment by gazzmaz4** [comment]
> > Make hivemind even more annoying than it currently is
> 
> I don't know how you managed it, but you managed it. Not a good or interesting ability, equivalent to having a free 'blind these people' button on a caste that you can't fight back against.

This super fucking good for benos. enables ambush tactics something benos need instead of just lee funni push denial.

**Comment by AyyRobotics** [comment]
Seems balanced enough (maybe would work better with a hivemind rework like others have said), and a very interesting concept that could lead to some cinematic horror experiences. Not bad

**Comment by BraveMole** [comment]
I'm giving up on this, splinter will never do his PR so whatever

**Comment by Moggram** [comment]
sad, maybe one day.

**Pull request closed without merging**